tokio/src/fs/file.rs

```diff
@@ -353,10 +353,10 @@ impl File {
             _ => unreachable!(),
         };
 
-        let seek = if !buf.is_empty() {
-            Some(SeekFrom::Current(buf.discard_read()))
-        } else {
+        let seek = if buf.is_empty() {
             None
+        } else {
+            Some(SeekFrom::Current(buf.discard_read()))
         };
```

Comment:
I'm not such a big fan of reversing the conditions. Some of them are fine, but in many cases, they are the way they are because the first condition is the most important case.